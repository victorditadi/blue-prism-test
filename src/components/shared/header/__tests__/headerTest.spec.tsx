import { render } from '@testing-library/react';

import { Header } from '..';

import * as nextRouter from 'next/router';

describe('Component: Header', () => {
  it('should render Header component', () => {
    expect(Header).toMatchSnapshot();
  });

  it('should contain Title', () => {
    const { queryByTestId } = render(<Header />);
    expect(queryByTestId('header-title')).toBeTruthy();
  });

  it('should contain MenuIcon', () => {
    const { queryByTestId } = render(<Header />);
    expect(queryByTestId('header-menu')).toBeTruthy();
  });

  it('should contain Schedules text inside Header Title /', () => {
    const { queryByTestId } = render(<Header />);
    const headerTitle = queryByTestId('header-title');

    expect(headerTitle?.textContent).toStrictEqual('Schedules');
  });

  it('should contain Schedules text inside Header Title /', () => {
    // Mock NextJS Use Router
    // @ts-ignore
    nextRouter.useRouter = jest
      .fn()
      .mockImplementation(() => ({ route: '/test', pathname: '/test' }));

    const { queryByTestId } = render(<Header />);
    const headerTitle = queryByTestId('header-title');

    expect(headerTitle?.textContent).toStrictEqual('Default');
  });
});
