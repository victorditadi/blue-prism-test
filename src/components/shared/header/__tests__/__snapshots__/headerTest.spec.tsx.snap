// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render Header component 1`] = `[Function]`;
